### ISSUE TITLE
SourceMaps lines offset for interpolated selectors 

### ISSUE THREAD

**Issue by rodneyrehm** [sass/libsass#2224]
The SourceMaps generation seems to trip over multi-line selectors if they contain interpolations (`#{$variable}`). In the demo SCSS below the generated CSS `.zulu-delta` is referenced to line 17, while it should be line 19. This does not happen for non-interpolated multi-line selectors. A single interpolation in the selector string causes the lines to collapse before the mutation could be tracked in SourceMaps. The offset is not recovered from, so all SCSS content *after* such an interpolated selector is affected by the shift in line number.

input.scss ([SourceMap](http://libsass.ocbnet.ch/srcmap/#JHByZWZpeDogenVsdTsKCi5hbHBoYSwKLmJyYXZvLAouY2hhcmxpZSB7CiAgY29sb3I6IHJlZDsKfQoKLmRlbHRhIHsKICBjb2xvcjogYmx1ZTsKfQoKLiN7JHByZWZpeH0tYWxwaGEsCi4jeyRwcmVmaXh9LWJyYXZvLAouI3skcHJlZml4fS1jaGFybGllIHsKICBjb2xvcjogcmVkOwp9CgouI3skcHJlZml4fS1kZWx0YSB7CiAgY29sb3I6IGJsdWU7Cn0KCi5lY2hvIHsKICBjb2xvcjogZ3JlZW47Cn0=))
```scss
$prefix: zulu;

.alpha,
.bravo,
.charlie {
  color: red;
}

.delta {
  color: blue;
}

.#{$prefix}-alpha,
.#{$prefix}-bravo,
.#{$prefix}-charlie {
  color: red;
}

.#{$prefix}-delta {
  color: blue;
}

.echo {
  color: green;
}
```

version info:
```cmd
$ node-sass --version
node-sass	3.11.0	(Wrapper)	[JavaScript]
libsass  	3.3.6	(Sass Compiler)	[C/C++]
```

**Comment by mgreter** [comment]
Thanks for the thorough bug report. I can see the issue clearly in the sourcemap inspector.

**Comment by mgreter** [comment]
@rodneyrehm can you tell on which versions you saw this error. Wondering if it got introduces with some latest sourcemap changes in 3.4-RC1. I specially would like to know if 3.3.6 already had this issue!? Thx!

**Comment by rodneyrehm** [comment]
>  can you tell on which versions you saw this error.

@mgreter the OP includes the `node-sass --version` output - this happened in libsass 3.3.6. I've also confirmed this issue still exists in libsass 3.4.0-RC1 via compiling it to [sass.js](https://github.com/medialize/sass.js).

**Comment by mgreter** [comment]
I have updated my [sourcemap inspector][1] to PR #2268. @nschonni can you check it to see if #2268 fixes the most urgent issue. As laid out in said PR, the mappings for the selector schema parts are still garbage, but at least to following mappings should no longer be off!

[1]: http://libsass.ocbnet.ch/srcmap/#JHByZWZpeDogenVsdTsKCi5hbHBoYSwKLmJyYXZvLAouY2hhcmxpZSB7CiAgY29sb3I6IHJlZDsKfQoKLmRlbHRhIHsKICBjb2xvcjogYmx1ZTsKfQoKLiN7JHByZWZpeH0tYWxwaGEsCi4jeyRwcmVmaXh9LWJyYXZvLAouI3skcHJlZml4fS1jaGFybGllIHsKICBjb2xvcjogcmVkOwp9CgouI3skcHJlZml4fS1kZWx0YSB7CiAgY29sb3I6IGJsdWU7Cn0KCi5lY2hvIHsKICBjb2xvcjogZ3JlZW47Cn0=

**Comment by mgreter** [comment]
I have merged #2268. It should fix the most urgent problems here!
